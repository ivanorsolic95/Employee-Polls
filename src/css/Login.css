.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
}

h1 {
  font-size: 2rem;
  margin-bottom: 1rem;
}

form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 300px;
  padding: 2rem;
  background-color: #f5f5f5;
  border-radius: 10px;
  box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.2);
}

label {
  font-size: 1.2rem;
  margin-bottom: 0.5rem;
}

input {
  width: 100%;
  padding: 0.5rem;
  margin-bottom: 1rem;
  border: none;
  border-radius: 5px;
  box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.2);
}

button {
  width: 100%;
  padding: 0.5rem;
  margin-top: 1rem;
  border: none;
  border-radius: 5px;
  background-color: #0077cc;
  color: #fff;
  font-size: 1.2rem;
  cursor: pointer;
  transition: background-color 0.2s ease-in-out;
}

button:hover {
  background-color: #005fa3;
}

p {
  color: red;
  margin-bottom: 1rem;
}
