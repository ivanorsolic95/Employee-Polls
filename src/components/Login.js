import { handleLogin } from "../actions/authedUser";
import { useState, useEffect } from "react";
import { useDispatch } from "react-redux";
import { useNavigate, useLocation } from "react-router-dom";
import "../css/Login.css";

const Login = () => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [loginError, setLoginError] = useState(false);
  const { state } = useLocation();

  const navigate = useNavigate();
  const dispatch = useDispatch();

  const handleUsername = (event) => {
    const value = event.target.value;
    setUsername(value);
  };

  const handlePassword = (event) => {
    const value = event.target.value;
    setPassword(value);
  };

  const handleSubmit = async (event) => {
    event.preventDefault();
    if (username && password) {
      const success = await dispatch(handleLogin(username, password));
      if (success) {
        setUsername("");
        setPassword("");
        navigate(state?.path || "/");
      } else {
        setUsername("");
        setPassword("");
        setLoginError(true);
      }
    }
  };

  useEffect(() => {
    if (loginError) {
      const timeout = setTimeout(() => {
        setLoginError(false);
      }, 3000);
      return () => clearTimeout(timeout);
    }
  }, [loginError]);

  return (
    <div className="container">
      <h1>Login</h1>

      <form onSubmit={handleSubmit}>
        <label>Username</label>
        <input
          value={username}
          placeholder="Username"
          onChange={handleUsername}
          type="text"
        />
        <label>Password</label>
        <input
          value={password}
          onChange={handlePassword}
          type="password"
          placeholder="Password"
        />
        <button
          data-testid="login-button"
          type="submit"
          disabled={!username || !password}
        >
          Login
        </button>
        {loginError && <p>Wrong username or password, please try again!</p>}
      </form>
    </div>
  );
};

export default Login;
