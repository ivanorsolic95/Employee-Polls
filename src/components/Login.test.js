import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import Login from "./Login";

describe("Login", () => {
  const mockStore = configureStore([]);
  const store = mockStore({});

  it("matches the snapshot when component is rendered", () => {
    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
