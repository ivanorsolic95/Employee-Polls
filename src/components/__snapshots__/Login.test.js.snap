// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches the snapshot when component is rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h1>
          Login
        </h1>
        <form>
          <label>
            Username
          </label>
          <input
            placeholder="Username"
            type="text"
            value=""
          />
          <label>
            Password
          </label>
          <input
            placeholder="Password"
            type="password"
            value=""
          />
          <button
            data-testid="login-button"
            disabled=""
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h1>
        Login
      </h1>
      <form>
        <label>
          Username
        </label>
        <input
          placeholder="Username"
          type="text"
          value=""
        />
        <label>
          Password
        </label>
        <input
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          data-testid="login-button"
          disabled=""
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
